pkg/diff/diff.go

```diff
@@ -283,6 +285,28 @@ func removeWebhookMutation(predictedLive, live *unstructured.Unstructured, gvkPa
 	return &unstructured.Unstructured{Object: pl}, nil
 }
 
+// Appends the key fields like "name" to the set elements which have some entries keyed by those fields.
+// This is needed to merge properly, see https://github.com/argoproj/argo-cd/issues/20792.
+func appendKeyFields(set *fieldpath.Set) {
+	// Map is to deduplicate
```

Comment:
Can you expand the comment to explain why de-duplication is necessary? What goes wrong if we don't de-dupe?

Reply:
I ended up replacing with a list, since a small optimization of reducing a number of calls to Has below might be overshadowed by the extra cost for a string conversion.